2013-02-15  H.J. Lu  <hongjiu.lu@intel.com>

	PR ld/15141
	* plugin.c (plugin_notice): Also trace symbol from the IR bfd.

diff --git a/ld/plugin.c b/ld/plugin.c
index da99e77..435f466 100644
--- a/ld/plugin.c
+++ b/ld/plugin.c
@@ -962,14 +962,14 @@ plugin_notice (struct bfd_link_info *info,
     {
       bfd *sym_bfd;
 
-      /* No further processing if this def/ref is from an IR dummy BFD.  */
+      /* Nothing to do here if this def/ref is from an IR dummy BFD.  */
       if (is_ir_dummy_bfd (abfd))
-	return TRUE;
+	;
 
       /* Making an indirect symbol counts as a reference unless this
 	 is a brand new symbol.  */
-      if (bfd_is_ind_section (section)
-	  || (flags & BSF_INDIRECT) != 0)
+      else if (bfd_is_ind_section (section)
+	       || (flags & BSF_INDIRECT) != 0)
 	{
 	  if (h->type != bfd_link_hash_new)
 	    {
